lib/lin_api.ml

```diff
@@ -256,10 +254,10 @@ module MakeCmd (ApiSpec : ApiSpec) : Lin.CmdSpec = struct
                   Iter.map (fun args -> Args.Fn (a,args)) (gen_shrinker_of_desc fdesc_rem args)
                        | _ -> failwith "Fn/None: should not happen")
            | Some shrk ->
-               Iter.(function (Args.Fn (a,args)) ->
-                   (map (fun a -> Args.Fn (a,args)) (shrk a))
-                   <+>
+               Iter.(function (Args.Fn (a,args)) -> (* heuristic: shrink last args first *)
```

Comment:
Is there a best choice for the first argument to shrink?
(My intuition would be that the best choice will change from one function to another.)

Reply:
I agree, this is very much a heuristic. Tracing shrink-logs I encountered countless lines of
```ocaml
Bytes.blit_string "loooooooooooooooooooong-bytes-strink-taking-forever-to-reduce-char-by-char" large_pos bytes_t large_pos large_length
```
where the shrinker would spend many minutes getting the first argument shrunk to `"aaaaaaaaaaaaaaa....aaaa"`, to finally get to reducing the later `int` parameters - which triggered a gradual reduction of the `string` length to just `"aaaa"` or so!

From that I inferred that in practice it is probably more common to have such "index" parameters later in an argument list (`Array.set`, `List.nth`, ...). This seems to work reasonably well in practice - and was motivated by the above - but I don't claim any optimality. Of course the heuristic is trivially defeated by swapping argument order.
Finally I would argue that left-to-right is just as much of a heuristic as right-to-left :smile:

This makes me wonder whether we can phrase an 'alternating' shrinker that, that repeatedly swaps the `Iter.t` to try candidates from... :thinking:

Reply:
An alternating shrinker would be nice indeed! At least I would be curious to see if that matters ;-)